core-bundle/src/Resources/contao/library/Contao/DcaLoader.php

```diff
@@ -198,6 +201,39 @@ private function addDefaultLabels($blnNoCache)
 			unset($v);
 		}
 	}
+
+	/**
+	 * Sets the parent table for the current table, if enabled and not set.
+	 */
+	private function setDynamicPTable(): void
+	{
+		if (!($GLOBALS['TL_DCA'][$this->strTable]['config']['dynamicPtable'] ?? false) || !empty($GLOBALS['TL_DCA'][$this->strTable]['config']['ptable']) || !isset($GLOBALS['BE_MOD']))
+		{
+			return;
+		}
+
+		if (null !== ($do = Input::get('do')))
```

Comment:
This should be an early return.

```suggestion
		if (!$do = Input::get('do'))
		{
			return;
		}
```